src/sentry/deletions/defaults/project.py

```diff
@@ -5,14 +5,20 @@ class ProjectDeletionTask(ModelDeletionTask):
     def get_child_relations(self, instance):
         from sentry import models
         from sentry.discover.models import DiscoverSavedQueryProject
-        from sentry.incidents.models import IncidentProject
+        from sentry.incidents.models import AlertRule, IncidentProject
         from sentry.replays.models import ReplayRecordingSegment
         from sentry.snuba.models import QuerySubscription
 
         relations = [
             # ProjectKey gets revoked immediately, in bulk
             ModelRelation(models.ProjectKey, {"project_id": instance.id})
         ]
+        relations.append(
+            ModelRelation(
+                AlertRule,
+                {"snuba_query__subscriptions__project": instance, "include_all_projects": False},
```

Comment:
Just noting that I _was_ going to write a migration to delete the orphaned metric alerts from before this was merged, but it turns out it's not possible because the only way to do it would be to look up the project ID through `alertrule.snuba_query.subscriptions[0].project_id` and check that that _doesn't_ exist in a list of all project ids that are active (`ObjectStatus.VISIBLE`), but since we delete the `QuerySubscription` row in this file, it's not possible to look it up this way. The `deletedproject` table stores the project slug but not the ID, and that's the only piece of information we have. c'est la vie ¯\_(ツ)_/¯